packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,527 @@
+  allowReplicaAccess = true,
+  allowBundleAccess = false,
+}: {
+  surescripts: EnvConfig["surescripts"];
+  allowReplicaAccess?: boolean;
+  allowBundleAccess?: boolean;
+}): Record<string, string> {
+  return {
+    SURESCRIPTS_SFTP_HOST: surescripts?.surescriptsHost,
+    SURESCRIPTS_SFTP_SENDER_ID: surescripts?.surescriptsSenderId,
+    SURESCRIPTS_SFTP_SENDER_PASSWORD: surescripts?.secrets.SURESCRIPTS_SFTP_SENDER_PASSWORD,
+    SURESCRIPTS_SFTP_RECEIVER_ID: surescripts?.surescriptsReceiverId,
+    SURESCRIPTS_SFTP_PUBLIC_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PUBLIC_KEY,
+    SURESCRIPTS_SFTP_PRIVATE_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PRIVATE_KEY,
+    ...(allowReplicaAccess
+      ? { SURESCRIPTS_REPLICA_BUCKET_NAME: surescripts?.surescriptsReplicaBucketName }
+      : {}),
+    ...(allowBundleAccess
+      ? { SURESCRIPTS_BUNDLE_BUCKET_NAME: surescripts?.surescriptsBundleBucketName }
+      : {}),
+  };
+}
+
+interface SurescriptsNestedStackProps extends NestedStackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction?: SnsAction;
+  lambdaLayers: LambdaLayers;
+  surescriptsReplicaBucket: s3.Bucket;
+  surescriptsBundleBucket: s3.Bucket;
+}
+
+export class SurescriptsNestedStack extends NestedStack {
+  // SFTP commands
+  readonly connectSftpLambda: Lambda;
+  readonly synchronizeSftpLambda: Lambda;
+  // Data pipeline Lambdas
+  readonly sendPatientRequestLambda: Lambda;
+  readonly sendPatientRequestQueue: Queue;
+  readonly receiveVerificationResponseLambda: Lambda;
+  readonly receiveVerificationResponseQueue: Queue;
+  readonly receiveFlatFileResponseLambda: Lambda;
+  readonly receiveFlatFileResponseQueue: Queue;
+  // Pipeline storage locations
+  readonly surescriptsBundleBucket: s3.Bucket;
+  readonly surescriptsReplicaBucket: s3.Bucket;
+
+  constructor(scope: Construct, id: string, props: SurescriptsNestedStackProps) {
+    super(scope, id, props);
+
+    this.terminationProtection = true;
+
+    const surescriptsBundleBucket = new s3.Bucket(this, "SurescriptsBundleBucket", {
+      bucketName: props.config.surescripts?.surescriptsBundleBucketName,
+      publicReadAccess: false,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      versioned: true,
+    });
+    this.surescriptsBundleBucket = surescriptsBundleBucket;
+
+    const surescriptsReplicaBucket = new s3.Bucket(this, "SurescriptsReplicaBucket", {
+      bucketName: props.config.surescripts?.surescriptsReplicaBucketName,
+      publicReadAccess: false,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      versioned: true,
+    });
+    this.surescriptsReplicaBucket = surescriptsReplicaBucket;
+
+    const commonConfig = {
+      lambdaLayers: props.lambdaLayers,
+      vpc: props.vpc,
+      envType: props.config.environmentType,
+      sentryDsn: props.config.lambdasSentryDSN,
+      alarmAction: props.alarmAction,
+      surescripts: props.config.surescripts,
+    };
+
+    // Only scoped to read/write from the S3 bucket
```

Comment:
What is this comment related to? I don't see anything related to S3 on that function.

Let's only have comments if we need to explain why something is being done, please. Otherwise, good names and simple logic trump comments that gets outdated.